internal/provider/models/engine.go

```diff
@@ -305,6 +321,43 @@ func ConditionGroupsAttribute() schema.SetNestedAttribute {
 	}
 }
 
+func GroupingKeysAttribute() schema.SetNestedAttribute {
+	return schema.SetNestedAttribute{
+		MarkdownDescription: "The attributes to group alerts by",
+		Required:            true,
+		NestedObject: schema.NestedAttributeObject{
+			Attributes: map[string]schema.Attribute{
+				"id": schema.StringAttribute{
```

Comment:
Kinda wish this was called `alert_attribute_id`, since it currently reads like the grouping key's own ID?

Also not convinced this func should live in `engine.go` rather than `alert_routes.go` or something like that - can't see it being reused in the medium term